doc/api-requests.md

```diff
@@ -0,0 +1,156 @@
+
+Releases Index
+--------------
+
+> **dandidav path:** `/dandisets/{dandiset_id}/releases/`
+
+For deep requests, `dandidav` paginates over the API's `/dandisets/{dandiset_id}/versions/` endpoint.
+
+For shallow requests, `dandidav` does not make any API requests.
+
+Top Level of a Dandiset Version
+-------------------------------
+
+> **dandidav paths:**
+>
+> - `/dandiset/{dandiset_id}/draft/`
+> - `/dandiset/{dandiset_id}/latest/`
+> - `/dandiset/{dandiset_id}/releases/{version_id}/`
+
+For both deep and shallow requests, if the request path is
+`/dandiset/{dandiset_id}/latest/`, an initial request is made to
+`/dandiset/{dandiset_id}/` to get the version ID of the latest published
+version.
+
+For deep requests, `dandidav` makes API requests to the following endpoints:
+
+- `/dandisets/{dandiset_id}/versions/{version_id}/info/`
+- `/dandisets/{dandiset_id}/versions/{version_id}/assets/paths/` (paginated)
+    - For each separate asset returned from this endpoint, `dandidav` makes
+      another request to
+      `/dandisets/{dandiset_id}/versions/{version_id}/assets/{asset_id}/info/`
+      to fetch further details about the asset.
+- `/dandisets/{dandiset_id}/versions/{version_id}/` (to fetch the version
+  metadata so that its size can be reported)
+
+For shallow requests, `dandidav` makes an API request to
+`/dandisets/{dandiset_id}/versions/{version_id}/info/`.
+
+Metadata File
+-------------
+
+> **dandidav paths:**
+>
+> - `/dandiset/{dandiset_id}/draft/dandiset.yaml`
+> - `/dandiset/{dandiset_id}/latest/dandiset.yaml`
+> - `/dandiset/{dandiset_id}/releases/{version_id}/dandiset.yaml`
+
+For both deep and shallow requests, if the request path is
+`/dandiset/{dandiset_id}/latest/`, an initial request is made to
+`/dandiset/{dandiset_id}/` to get the version ID of the latest published
+version.  Then, for both deep and shallow requests, `dandidav` makes an API
+request to `/dandisets/{dandiset_id}/versions/{version_id}/`.
+
+Asset Path
+----------
+
+> **dandidav paths:**
+>
+> - `/dandiset/{dandiset_id}/draft/{path}`
+> - `/dandiset/{dandiset_id}/latest/{path}`
+> - `/dandiset/{dandiset_id}/releases/{version_id}/{path}`
+>
+> Note that any trailing slashes at the end of `path` are ignored and are
+> stripped before passing to the Archive.
+
+For both deep and shallow requests, if the request path is
+`/dandiset/{dandiset_id}/latest/`, an initial request is made to
+`/dandiset/{dandiset_id}/` to get the version ID of the latest published
+version.
+
+Then, for each initial subpath of `path` that ends with either (a) a component
+ending in ".zarr" or ".ngff" (case insensitive) or (b) the end of the path, an
+API request is made to
+`/dandisets/{dandiset_id}/versions/{version_id}/assets/?path={subpath}&metadata=1&order=path`,
+which is paginated through until one of the following occurs:
+
+- An asset whose path equals `subpath` is found.  In this case:
```

Comment:
since it is just "prefix" not really a "path"

might be worth referencing that 
- https://github.com/dandi/dandi-archive/issues/1562

and shouldn't we strive to fix this -- sounds like another inefficiency.  Would the need for it be avoided by 

- https://github.com/dandi/dandi-archive/issues/1837

?